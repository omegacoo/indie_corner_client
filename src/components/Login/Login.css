.loginContainer {
    position: relative;
    display: block;
    border: 5px solid white;
    border-radius: 5px;
    background-color: black;
    width: 40vw;
    height: fit-content;
    padding: 2vh;
    z-index: 2;
    margin-left: 35vw;
    margin-top: -22.5vh;
    z-index: 3;
}

.loginContainer input {
    width: 90%;
}

.closeLogin {
    position: absolute;
    background-color: black;
    color: white;
    border: none;
    right: 0;
    top: 0;
    font-size: large;
}

.login_button {
    display: block;
    margin: 1.5vh;
    margin-left: auto;
    margin-right: auto;
    background-color: black;
    border: 5px solid white;
    border-radius: 5px;
    color: white;
    padding: 1vh;
}

#Login_error {
    color: red;
    font-weight: normal;
}