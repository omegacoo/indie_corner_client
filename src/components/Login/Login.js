import React from 'react';

import StoreContext from '../../StoreContext';

import './Login.css';

export default class Login extends React.Component {
    static contextType = StoreContext;

    state = {
        username: '',
        password: ''
    }

    handleUsernameChange = e => {
        this.setState({
            username: e.target.value
        })
    }

    handlePasswordChange = e => {
        this.setState({
            password: e.target.value
        })
    }

    handleLoginSubmit = e => {
        e.preventDefault();
        
        this.context.handleLogin(this.state.username, this.state.password);
    }

    render(){
        return(
            <div className='loginContainer'>
                <span className='popup'>
                    <button
                        onClick={ this.props.handleClose }
                        className='closeLogin'
                    >
                        X
                    </button>
                    <form 
                        className='login_form'
                        onSubmit={this.handleLoginSubmit}
                    >
                        <label htmlFor='login_username'>Username: </label>
                        <input 
                            value={this.state.username}
                            onChange={this.handleUsernameChange}
                            type='text'
                            autoFocus
                        />
                        <label htmlFor='login_password'>Password: </label>
                        <input
                            value={this.state.password}
                            onChange={this.handlePasswordChange}
                            type='password'
                        />
                        {this.state.error ? <h3 id='Login_error'>{this.state.error}</h3> : null}
                        <button className='login_button' type='submit'>login</button>
                    </form>
                </span>
            </div>
        );
    }
}