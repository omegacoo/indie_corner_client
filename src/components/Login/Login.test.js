import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Login from './Login';

describe('Login component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');

        ReactDOM.render(<Login />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders UI correctly', () => {
        expect(
        renderer.create(<Login />).toJSON()
        ).toMatchSnapshot();
    });
});