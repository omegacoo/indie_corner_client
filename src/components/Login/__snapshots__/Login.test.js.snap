// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders UI correctly 1`] = `
<div
  className="loginContainer"
>
  <span
    className="popup"
  >
    <button
      className="closeLogin"
    >
      X
    </button>
    <form
      className="login_form"
      onSubmit={[Function]}
    >
      <label
        htmlFor="login_username"
      >
        Username: 
      </label>
      <input
        autoFocus={true}
        onChange={[Function]}
        type="text"
        value=""
      />
      <label
        htmlFor="login_password"
      >
        Password: 
      </label>
      <input
        onChange={[Function]}
        type="password"
        value=""
      />
      <button
        className="login_button"
        type="submit"
      >
        login
      </button>
    </form>
  </span>
</div>
`;
